### PR TITLE
New resource `azurerm_data_share_dataset_data_lake_gen1`

### PR DESCRIPTION
**Pull request by yupwei68**
Partially fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/6480

Sub PR of : https://github.com/terraform-providers/terraform-provider-azurerm/pull/7064

=== RUN   TestAccAzureRMDataShareDataSetDataLakeGen1File_basic
=== PAUSE TestAccAzureRMDataShareDataSetDataLakeGen1File_basic
=== CONT  TestAccAzureRMDataShareDataSetDataLakeGen1File_basic
--- PASS: TestAccAzureRMDataShareDataSetDataLakeGen1File_basic (395.53s)
=== RUN   TestAccAzureRMDataShareDataSetDataLakeGen1Folder_basic
=== PAUSE TestAccAzureRMDataShareDataSetDataLakeGen1Folder_basic
=== CONT  TestAccAzureRMDataShareDataSetDataLakeGen1Folder_basic
--- PASS: TestAccAzureRMDataShareDataSetDataLakeGen1Folder_basic (335.91s)
=== RUN   TestAccAzureRMDataShareDataSetDataLakeGen1_requiresImport
=== PAUSE TestAccAzureRMDataShareDataSetDataLakeGen1_requiresImport
=== CONT  TestAccAzureRMDataShareDataSetDataLakeGen1_requiresImport
--- PASS: TestAccAzureRMDataShareDataSetDataLakeGen1_requiresImport (410.60s)

### CODE REVIEW

File: website/docs/r/data_share_dataset_data_lake_gen1.html.markdown [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    azurerm_role_assignment.example,
+  ]
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Data Share Data Lake Gen1 Dataset. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.
```
Comment (WodansSon) [inline website/docs/r/data_share_dataset_data_lake_gen1.html.markdown:83]:
```suggestion
* `name` - (Required) The name of the Data Share Data Lake Gen1 Dataset. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.

```

File: website/docs/r/data_share_dataset_data_lake_gen1.html.markdown [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+* `data_share_id` - (Required) The ID of the Data Share in which this Data Share Data Lake Gen1 Dataset should be created. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.
```
Comment (WodansSon) [inline website/docs/r/data_share_dataset_data_lake_gen1.html.markdown:85]:
```suggestion
* `data_share_id` - (Required) The resource ID of the Data Share where this Data Share Data Lake Gen1 Dataset should be created. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.

```

File: website/docs/r/data_share_dataset_data_lake_gen1.html.markdown [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+* `data_lake_store` - (Required) A `data_lake_store` block as defined below.
+
+* `folder_path` - (Required) The folder path of the data lake store to be shared with the receiver. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.
+
+* `file_name` - (Optional) The file name in the folder path of the data lake store to be shared with the receiver. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.
```
Comment (WodansSon) [inline website/docs/r/data_share_dataset_data_lake_gen1.html.markdown:91]:
```suggestion
* `file_name` - (Optional) The file name of the data lake store to be shared with the receiver. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.

```

File: website/docs/r/data_share_dataset_data_lake_gen1.html.markdown [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+* `resource_group_name` - (Required)  The resource group name of the data lake store to be shared with the receiver. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.
+
+* `subscription_id` - (Required) The subscription id of the data lake store to be shared with the receiver. Changing this forces a new Data Share Data Lake Gen1 Dataset to be created.
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the Data Share Data Lake Gen1 Dataset.
```
Comment (WodansSon) [inline website/docs/r/data_share_dataset_data_lake_gen1.html.markdown:107]:
```suggestion
* `id` - The resource ID of the Data Share Data Lake Gen1 Dataset.

```

File: website/docs/r/data_share_dataset_data_lake_gen1.html.markdown [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+* `display_name` - The name of the Data Share Dataset.
```
Comment (WodansSon) [inline website/docs/r/data_share_dataset_data_lake_gen1.html.markdown:109]:
```suggestion
* `display_name` - The displayed name of the Data Share Dataset.

```

File: azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datashare/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datashare/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataShareDataSetDataLakeGen1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataShareDataSetDataLakeGen1Create,
```
[the remark this answers is not in the document]
Comment (yupwei68) [reply azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go:24]: No, there is no `Update`

**Comment by yupwei68** [comment]
Hi @WodansSon Thanks for your comments. Corresponding changes have been pushed. Please continue reviewing.

File: azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+						"subscription_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validation.IsUUID,
+						},
+					},
+				},
+			},
```
Comment (katbyte) [inline azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go:80]: is there any reason we can't have this as `data_lake_store_id` and grab the 3 properties from it? if not i think we should do that instead for better ux

Comment (yupwei68) [reply azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go:80]: Hi kt, it for the `data_lake_store_id` could be subscription crossed , so we could not get the `data_lake_store_id` by its client, and we shall concatenate the id if we want only one field `data_lake_store_id`.

Comment (katbyte) [reply azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go:80]: Could you not get the subscription from the id? `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DataLakeStore/accounts/mydatalakeaccount`

Comment (katbyte) [reply azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go:80]: I think all the required bits on info are there?

**Comment by yupwei68** [comment]
hi @katbyte Thanks for your comments. Corresponding changes have been pushed. Please continue reviewing.

**Comment by WodansSon** [comment]
image

**Pull request merged**